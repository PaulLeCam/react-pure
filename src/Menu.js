import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

export default class PureMenu extends Component {
  static propTypes = {
    horizontal: PropTypes.bool,
    scrollable: PropTypes.bool,
  };

  static defaultProps = {
    horizontal: false,
    scrollable: false,
  };

  render() {
    const { className, horizontal, scrollable, ...props } = this.props;
    const cls = classNames(
      'pure-menu',
      className,
      horizontal && 'pure-menu-horizontal',
      scrollable && 'pure-menu-scrollable'
    );

    return <div className={cls} {...props} />;
  }
}
