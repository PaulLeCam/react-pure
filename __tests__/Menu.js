import React from 'react';
import TestRenderer from 'react-test-renderer';

import Menu from '../src/Menu';

describe('Menu', () => {
  it('renders a div with class `pure-menu` by default', () => {
    const renderer = TestRenderer.create(<Menu className="my-menu" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-menu-horizontal` class if the `horizontal` property is set', () => {
    const renderer = TestRenderer.create(<Menu horizontal />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-menu-scrollable` class if the `scrollable` property is set', () => {
    const renderer = TestRenderer.create(<Menu scrollable />);
    expect(renderer.toJSON).toMatchSnapshot();
  });
});
