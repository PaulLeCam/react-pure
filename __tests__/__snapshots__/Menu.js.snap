// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu adds the \`pure-menu-horizontal\` class if the \`horizontal\` property is set 1`] = `[Function]`;

exports[`Menu adds the \`pure-menu-scrollable\` class if the \`scrollable\` property is set 1`] = `[Function]`;

exports[`Menu renders a div with class \`pure-menu\` by default 1`] = `[Function]`;
